test/Archive/ArchiveFile.test.cpp

```diff
@@ -0,0 +1,69 @@
+#include "Archive/VolFile.h"
+#include "Archive/ClmFile.h"
+#include "XFile.h"
+#include <gtest/gtest.h>
+#include <vector>
+#include <string>
+
+void TestEmptyArchive(Archive::ArchiveFile& archiveFile, const std::string& archiveFilename);
+
+
+TEST(VolFile, EmptyArchive) 
+{
+	const std::string archiveFilename("EmptyArchive.vol");
+
+	{
+		// Create an empty archive
+		const std::vector<std::string> filesToPack;
+		Archive::VolFile::CreateArchive(archiveFilename, filesToPack);
+
+		Archive::VolFile archiveFile(archiveFilename);
+
+		SCOPED_TRACE("Empty Volume File");
+		TestEmptyArchive(archiveFile, archiveFilename);
+
+		EXPECT_THROW(archiveFile.GetCompressionCode(0), std::runtime_error);
```

Comment:
I don't know if this check is really needed. It seems like a roundabout and less clear way of testing `EXPECT_EQ(0, archiveFile.GetCount());`, which is already checked.

Reply:
The goal was to ensure that when you tried to pull a compression code from an empty volume, the proper error was thrown. If you think it is too redundant I suppose I don't mind if it is stricken.